src/noisepy/seis/channelcatalog.py

```diff
@@ -54,25 +55,27 @@ def get_inventory(self, timespan: DateTimeRange, station: Station) -> obspy.Inve
 
 class XMLStationChannelCatalog(ChannelCatalog):
     """
-    A channel catalog that reads <station>.XML files from a directory
+    A channel catalog that reads <station>.XML files from a directory or an s3://... bucket path.
     """
 
-    def __init__(self, xmldir: str) -> None:
+    def __init__(self, xmlpath: str) -> None:
         super().__init__()
-        self.xmldir = xmldir
-        if not os.path.exists(self.xmldir):
-            raise Exception(f"The XML Station file directory '{xmldir}' doesn't exist")
+        self.xmlpath = xmlpath
+        self.fs = get_filesystem(xmlpath)
+        if not self.fs.exists(self.xmlpath):
+            raise Exception(f"The XML Station file directory '{xmlpath}' doesn't exist")
 
     def get_inventory(self, timespan: DateTimeRange, station: Station) -> obspy.Inventory:
-        xmlfile = os.path.join(self.xmldir, f"{station.network}_{station.name}.xml")
+        xmlfile = os.path.join(self.xmlpath, f"{station.network}_{station.name}.xml")
```

Comment:
This assumes that `xmlpath` is a directory path. Maybe in `__init__` you there need to be a check for this also, otherwise, you might get something like `s3://mybucket/path/myxml.xml` which would cause this join to give you the wrong path to actual `xml` file.